syntax = "proto3";

package lists.book;

import "google/protobuf/timestamp.proto";

option cc_enable_arenas = true;

enum ReadingStatus {
    READING_STATUS_UNSPECIFIED = 0;
    READING_STATUS_NOT_STARTED = 1;
    READING_STATUS_READING = 2;
    READING_STATUS_FINISHED = 3;
}

enum Rating {
  RATING_UNSPECIFIED = 0;
  RATING_1 = 1;
  RATING_2 = 2;
  RATING_3 = 3;
  RATING_4 = 4;
  RATING_5 = 5;
}

message Book {
    int64 id = 1;
    string title  =2;
    string cover_image_url = 3;
    string author = 4;
    google.protobuf.Timestamp startDate = 5;
    google.protobuf.Timestamp endDate = 6;
    Rating rating = 7;
    ReadingStatus status = 8;
    string note = 9;

    google.protobuf.Timestamp created_at  = 20;
    google.protobuf.Timestamp updated_at  = 21;
}

message BookCard {
    int64 id = 1;
    string title  =2;
    string cover_image_url = 3;
    ReadingStatus status = 4;
}